gateway/gateway.go

```diff
@@ -693,6 +701,66 @@ func (gateway *HandleT) getJobDataFromRequest(req *webRequestT) (jobData *jobFro
 	return
 }
 
+func setBatch(body []byte, reqType string) (batch []byte, err error) {
+	body, err = sjson.SetBytes(body, "type", reqType)
+	if err != nil {
+		err = errors.New(response.NotRudderEvent)
+		return
+	}
+	batch, err = sjson.SetRawBytes(BatchEvent, "batch.0", body)
+	if err != nil {
+		err = errors.New(response.NotRudderEvent)
+		return
+	}
+	return
+}
+
+func isNonIdentifiable(anonIDFromReq, userIDFromReq string) bool {
+	if anonIDFromReq == "" && userIDFromReq == "" {
+		return !allowReqsWithoutUserIDAndAnonymousID
+	}
+	return false
+}
+
+func buildUserID(userIDHeader, anonIDFromReq, userIDFromReq string) string {
+	if anonIDFromReq != "" {
+		return userIDHeader + DELIMITER + anonIDFromReq + DELIMITER + userIDFromReq
+	}
+	return userIDHeader + DELIMITER + userIDFromReq + DELIMITER + userIDFromReq
+}
+
+func (gateway *HandleT) isUserSuppressed(workspaceID, userID, sourceID string) bool {
+	if !enableSuppressUserFeature || gateway.suppressUserHandler == nil {
+		return false
+	}
+	return gateway.suppressUserHandler.IsSuppressedUser(
+		workspaceID,
+		userID,
+		sourceID,
+	)
+}
+
+// checks for the presence of messageId in the event
+//
+// sets to a new uuid if not present
+func checkMessageID(event map[string]interface{}) {
```

Comment:
Might make sense to rename this function. It's not just checking but setting a key on the given map.

```suggestion
func setRandomMessageIDWhenEmpty(event map[string]interface{}) {
```